jujugui/static/gui/src/app/app.js

```diff
@@ -2255,6 +2278,22 @@ YUI.add('juju-gui', function(Y) {
     },
 
     /**
+      Calls the necessary methods to setup the GUI and put the user in the
+      Deployment Flow when they have used Direct Deploy.
+
+      @param {Object} dd - The Direct Deploy data from state.
+    */
+    _directDeploy: function(ddData) {
+      this.state.changeState({
+        special: {dd: null}
+      });
+      // Reusing deployTarget state method so faking request signature
```

Comment:
I updated it to be: 

      // The deployTarget method is called directly by state. We're reusing it
      // here so the state call signature needs to be replicated.